app/uk/gov/hmrc/lisaapi/controllers/BulkPaymentController.scala

```diff
@@ -38,33 +38,54 @@ class BulkPaymentController @Inject()(
+                        periodEndDate: Option[String] = None
+                      ): Action[AnyContent] =
     validateHeader().async { implicit request =>
       implicit val startTime: Long = System.currentTimeMillis()
       withValidLMRN(lisaManager) { () =>
         withEnrolment(lisaManager) { _ =>
-          withValidDates(startDate, endDate) { (start, end) =>
-            val response = service.getBulkPayment(lisaManager, start, end)
-
-            response flatMap {
-              case s: GetBulkPaymentSuccessResponse => {
-                lisaMetrics.incrementMetrics(startTime, OK, LisaMetricKeys.TRANSACTION)
-                withApiVersion {
-                  case Some(VERSION_1) => Future.successful(transformV1Response(Json.toJson(s)))
-                  case Some(VERSION_2) => Future.successful(Ok(Json.toJson(s)))
+          withApiVersion {
+            case Some(VERSION_1) => withValidDates(startDate, endDate) { (periodStart, periodEnd) =>
+              val response = service.getBulkPayment(lisaManager, periodStart, periodEnd)
+
+              response flatMap {
+                case s: GetBulkPaymentSuccessResponse => {
+                  lisaMetrics.incrementMetrics(startTime, OK, LisaMetricKeys.TRANSACTION)
+                  Future.successful(transformV1Response(Json.toJson(s)))
                 }
-              }
-              case GetBulkPaymentNotFoundResponse => {
-                lisaMetrics.incrementMetrics(startTime, NOT_FOUND, LisaMetricKeys.TRANSACTION)
-                withApiVersion {
-                  case Some(VERSION_1) => Future.successful(NotFound(Json.toJson(ErrorBulkTransactionNotFoundV1)))
-                  case Some(VERSION_2) => Future.successful(NotFound(Json.toJson(ErrorBulkTransactionNotFoundV2)))
+                case GetBulkPaymentNotFoundResponse => {
+                  lisaMetrics.incrementMetrics(startTime, NOT_FOUND, LisaMetricKeys.TRANSACTION)
+                  Future.successful(NotFound(Json.toJson(ErrorBulkTransactionNotFoundV1)))
+                }
+                case GetBulkPaymentServiceUnavailableResponse => Future.successful(ErrorServiceUnavailable.asResult)
+                case _ => {
+                  lisaMetrics.incrementMetrics(startTime, INTERNAL_SERVER_ERROR, LisaMetricKeys.TRANSACTION)
+                  Future.successful(InternalServerError(Json.toJson(ErrorInternalServerError)))
                 }
               }
-              case GetBulkPaymentServiceUnavailableResponse => Future.successful(ErrorServiceUnavailable.asResult)
-              case _ => {
-                lisaMetrics.incrementMetrics(startTime, INTERNAL_SERVER_ERROR, LisaMetricKeys.TRANSACTION)
-                Future.successful(InternalServerError(Json.toJson(ErrorInternalServerError)))
+            }
+            case Some(VERSION_2) => withValidDates(periodStartDate, periodEndDate) { (periodStart, periodEnd) =>
+              val response = service.getBulkPayment(lisaManager, periodStart, periodEnd)
```

Comment:
This looks like the same code as above (unless I'm missing something)? Can this be refactored?